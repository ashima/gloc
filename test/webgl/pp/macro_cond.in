#define Q
#line 100
#define J defined Q
#if J
void foo() {}
#else
void bad_foo() {}
#endif
#undef Q
#if !J
void remote_defined() {}
#else
void bad_remote_defined() {}
#endif

#line 1000
#define A 1
#define B 2
#define OP defined
#if OP A && OP B
#define Q 7
#endif
int blah() { return Q; }
#if OP W
#define Q 77
#endif
void main() { gl_Position = Q; }
#define defined Q
#if defined < 100
int baz() { return defined; }
#endif
