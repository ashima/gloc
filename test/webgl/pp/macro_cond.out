#line 102
void foo() {}





void remote_defined() {}
#line 1006
int blah() { return
#line 1004
          7

                     ; }



void main() { glPosition =
#line 1004
          7





                            ; }


int baz() { return
#line 1004
          7








                          ; }
